### ISSUE TITLE
Feature Request - Document procedures

### ISSUE THREAD

**Issue by sylvainar** [legoboyvdlp/A320-family#135]
**Is your feature request related to a problem or missing feature? Please describe.**
Hello there, I'm trying to go further in realism and stop using autostart on airplanes I'm using in FlightGear. I don't understand why this airplane doesn't have an interactive checklist like others (C172, CR700 or EC135 for instance). The A320 family is awesome in term of realism and completeness, there is so much work behind it, but newcomers (like me lol) have a hard time figuring how to start the airplane manually. I ended up watching some youtube videos because I didn't find any recent information on wikis, doc or forums.

**Describe the solution you'd like**
I've never contributed to flightgear before and I don't know how much work it is to integrate interactive checklists to airplanes.
If it's normal that it's not done in the airplane directly, maybe write a wiki page with the procedure.

**Comment by legoboyvdlp** [comment]
Yeah, interactive checklists are possible. For instance, 

![image](https://user-images.githubusercontent.com/10434911/81496996-823c1b80-92b3-11ea-8417-49a62c40291f.png)


But, checklists I really think are for checklists, not really "procedures" to start, etc.

![image](https://user-images.githubusercontent.com/10434911/81497027-aef03300-92b3-11ea-9764-1479fcf39319.png)

These are the real Airbus checklists. They are not really to tell you how to start up, but to remind you in case you forgot something like flaps :)

Maybe, I could make some checklists and make it clear that they are "procedure" checklists, not "proper" checklists. 

I definitely intended to make a full manual in the end.

**Comment by sylvainar** [comment]
Why not both? Interactive checklist could be use for startup procedures, since you already propose a "condensed" checklist to print on PDF, for those who want to push the simulation further and only do it with the card.

Currently, when you start the engines, you do the procedure by memory or do you have it documented somewhere?

**Comment by Octal450** [comment]
Yes, on airliners (unlike GA) the checklists don't actually teach you how to start/use the thing. There is a plan to add a tutorial system that will walk you through this however, but not yet as this means any time we change something we have to worry about the "tutorials" causing problems. Best to leave this to when the aircraft is mostly complete.

You can follow a FSLabs A320 tutorial on YouTube, which will teach you how to start it, as both them and us are modelling based on real aircraft manuals.

Kind Regards,
Josh

**Comment by sylvainar** [comment]
Okay, fair enough. I just found the subject interesting and wanted to get involved, if this is bad timing that's okay.

Linking here the resources I used for further reference:
- https://www.youtube.com/watch?v=UFcz8WrOR78
- http://wiki.flightgear.org/Fr/Airbus_A320_Family (in french, outdated)

**Comment by merspieler** [comment]
In fact, I've started already on tutorials a while ago... but, as @Octal450 said, every time we change something, we've got to check/update the tutorial.

As well, the systems are so complex... that a full, complete tutorial for startup might end up being like 30min.... so it's not easy to find a balance

**Comment by Octal450** [comment]
@sylvainar we appreciate the suggestions!

Kind Regards,
Josh

**Comment by dirteat** [comment]
Yep, I'll vote for that one :)

I understand it is far from being a priority, but it just misses almost nothing to allow anyone to jump on it and start.

I've just made my maiden flight today, and the parts I did not guess alone, or with insim info, were just, in order of importance:

1) APU bleed ON, before Ignit Knob ON and Engine 1/2 starts. You can guess it if you look carefully to the arrow directions in the AIR COND section, especially the "start" label, but quite tough to reverse engineer!

2) ADIRS of course, especially the default real setting that makes them aligning after 10 of minutes. There is no way a newcomer would find this out alone. So, I would suggest making the "Instant alignment" the default, possibly with a online message for the purists.

3) INIT data for making a basic flight plan on the onboard computer (don't remember the official name of it), which is superbly well rendered by the way!!

PS: You can actually fly with 1) only, which is also kind of cool as this is the only way to do aerobatics with the A320.

**Comment by merspieler** [comment]
>  So, I would suggest making the "Instant alignment" the default

Won't be happening... we aim at study level so it's kinda expected to know the aircraft.

If you are a new user and just want to fly, you can use the ready for takeoff state, available via aircraft config.

**Comment by dirteat** [comment]
No pb to me, it doesn't hurt to add somewhere that the ADRIS takes 10mn to be in stage though. Unless of course, you really want to scare all newcomers that did not read the full manual, but then, don't bother adding a checklist nor writing a wiki :)

**Comment by merspieler** [comment]
it doesn't take exactly 10 min, it depends on your location, it's faster near the equator and takes longer towards the poles.

Well... if they don't know about ADIRS they probably don't know about a LOT of other things either... so you've got to give them a crash course. There's nothing against adding a tutorial (fg tutorials can even highlight the switch or pb you need to press). It's just, that we're not yet ready to do this as internal things might change and the the tutorial would break which only causes double work.
So we've decided to leave it be and revisit it, once we're pretty much done.

If you want to make the effort of adding and maintaining the tutorial, you're more than welcome to contribute.
You can find my tutorial branch which I've started (and subsequently stopped after a similar discussion) back in the IDG times [here](https://gitea.merspieler.tk/fly/IDG-A32X/src/branch/tut)

**Comment by dirteat** [comment]
Thanks for the link, I'll try to contribute to the wiki or tutorial.

PS: I just wanted to add that, even though new comers can jump and fly with auto-start, there is this incontestable pleasure to start the engines and hearing their revs-up. Especially when the sounds are so well rendered, and match the real engines, as they are in the 320.

**Comment by merspieler** [comment]
> there is this incontestable pleasure to start the engines and hearing their revs-up

yeah... see it as reward for learining how to start it up ;-)

**Comment by dg-505** [comment]
To revive this thread...

I just started implementing the SOP based on the FCOM as an interactive checklist. (Yes I know, a Pilot should know the SOP items from memory, but having them interactive is a good way to learn them)

As part of this, I've added some knob animations (Weather Radar controls, Switching Panel knobs, LDG ELEV knob).

In addition I started tryig to implement kind of a "Virtual F/O" who will also function as the PM. This guy will execute his part of the SOP/Tasksharing as required by the FCOM such as the Preliminary Cockpit Preparation, Exterior Walkaround and so on (sometimes when commanded by the PF (PM: "Positive Climb!" -> PF: "Gear up!" -> PM: Execute action and announce "Gear up") and, sometimes by himself, like After landing: switch of the Weather Radar, set TCAS to STBY, retract Flaps, Start the APU, activate Brake Fan and so on). This actions might be triggered by a command from the PF (keyboard shortcut => no view movement and mouse click neccessary) or automatically when the PF executed his actions, or a certain state is reached (e.g. after landing: WOW && <80kt && Strobes off && LDG lights off && Taxi light on)

What do you think about that?

**Comment by Octal450** [comment]
I think this should be developed as a framework that can apply to all airplanes and then aircraft can implement specific calls and such.

I cannot speak for other contributors but from my end, this project is on hold and assuming nobody else takes the lead, will be restructured heavily in the future. Please read this thread for more information: https://forum.flightgear.org/viewtopic.php?f=4&t=42416

Kind Regards,
Josh

**Comment by dg-505** [comment]
Thanks for the link. I totally missed that post on the forum.

I'm not sure how to understand this IntegratedSystems-thingy. Is it a theoretical concept of how to write clean aircraft code? Or is it some common code base that can be adapted to different aircraft? Is it available somwhere so that one can contribute to it instead of contributing to individual aircraft?

Regarding your three options, I would prefer option 1 or 2.
Though I'm not sure what would be the more intelligent way: Make the 320 IS-like and use this as a base for the A330, or redo the A330 and use it as a base for rewriting the 320...

Did you finally make a decision on how to progress? What would be the best way to contribute, if you choose option 1 or 2?
